### PR TITLE
:heavy_plus_sign: Add Deno adapter demo (#75)

### PR DESCRIPTION
**Pull request by siguici**
I just added a demo that uses [Astro's Deno adapter](https://github.com/denoland/deno-astro-adapter) and I encountered no errors either for the dev script or for the build script as mentioned in #75.

### CODE REVIEW

**Comment by thejackshelton** [comment]
Our main use case for process.env is to communicate between `server.ts` and `index.ts` in the integration.

If I recall Arsh from the core team mentioned something about a vite global that could be used instead a while back 🤔

File: apps/astro-deno-demo/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "scripts": {
+    "dev": "astro dev",
+    "start": "astro dev",
+    "build": "astro check && astro build",
+    "preview": "deno run --allow-net --allow-read --allow-env ./dist/server/entry.mjs",
+    "astro": "astro"
+  },
+  "dependencies": {
+    "@astrojs/check": "^0.5.10",
+    "@astrojs/deno": "^5.0.1",
```
Comment (Mac-Mann) [inline apps/astro-deno-demo/package.json:14]: https://github.com/denoland/deno-astro-adapter/issues/10#issuecomment-2016125546
https://github.com/denoland/deno-astro-adapter/issues/13

Not sure if you have seen either of these comments?

Given the outdated NPM registry, might want to change this to:
```suggestion
    "@astrojs/deno": "@astrojs/deno": "github:denoland/deno-astro-adapter",
```

**Comment by siguici** [comment]
> Our main use case for process.env is to communicate between `server.ts` and `index.ts` in the integration.

I'm aware of that, and I believe it's the only way at the moment to achieve this. Until we find a better solution.
 
> If I recall Arsh from the core team mentioned something about a vite global that could be used instead a while back 🤔

I know it's possible to use environment variables from Vite's user configuration, but I'm not sure if there's a way to pass them globally through Vite since it's intended for frontend use and not server-side. So I think it's possible to do it in the `src/index.ts` file, but for the `server.ts` file, we'll have to use the method suggested by the server-side runtime.

**Comment by thejackshelton** [comment]
Good catch @Mac-Mann! Also @siguici I reviewed the PR and it looks good! Maybe we should change the names of the demo's to: `deno-demo` and `node-demo`.

https://github.com/QwikDev/astro/pull/77#discussion_r1543129687

what do we think about the suggested change here? I've heard of the issues with Astro and Deno up to this point.

File: apps/astro-deno-demo/package.json (hunk 1 above)
Comment (siguici) [reply apps/astro-deno-demo/package.json:14]: > Hey @siguici Thanks for taking on this issue! I have been out sick for the past few days and am just getting caught up with this discussion.
> 
> Not sure if you have seen either of these comments?

Wishing you a speedy recovery.

I've just seen the issue, but I don't think the problem lies there, considering that Astro's integration with Deno doesn't seem to be an issue. The problem seems to be more related to the use of environment variables. Deno doesn't support process.env.
However, I'll try to explore this way to see if it resolves the problem.

**Comment by siguici** [comment]
> Good catch @Mac-Mann! Also @siguici I reviewed the PR and it looks good! Maybe we should change the names of the demo's to: `deno-demo` and `node-demo`.

Thank you. That's a very good suggestion.

Yes, I tried the same thing and it also failed.

I'm making the suggested changes in the PR.

**Comment by thejackshelton** [comment]
@siguici think it's good to merge? Looks good to me!

**Comment by siguici** [comment]
> @siguici think it's good to merge? Looks good to me!

Yes, it can be merged without any issues.

**Comment by thejackshelton** [comment]
Merged :)

**Pull request merged**